addons/mail/static/tests/composer/composer_tests.js

```diff
@@ -436,6 +436,41 @@ QUnit.test("pending mentions are kept when toggling composer", async (assert) =>
     assert.containsOnce($, ".o-mail-Message-body a.o_mail_redirect:contains(@Mitchell Admin)");
 });
 
+QUnit.test('suggestion list is switchable between commands', async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({
+        email: "testpartner@odoo.com",
+        name: "Luigi",
+    });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Mario Party",
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId }),
+            Command.create({ partner_id: partnerId }),
+        ],
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    assert.strictEqual($(".o-mail-Composer-input").val(), "");
+    await insertText(".o-mail-Composer-input", "#");
+    await contains(".o-mail-Composer-suggestion", { count: 1, text: "#Mario Party" });
+    await click(".o-mail-Composer-suggestion");
+    assert.strictEqual(
+        $(".o-mail-Composer-input").val().replace(/\s/, " "),
+        "#Mario Party ",
+        "previous content + mentioned channel + additional whitespace afterwards"
+    );
+    await insertText(".o-mail-Composer-input", "@");
+    await contains(".o-mail-Composer-suggestion", { count: 1, text: "Luigi" });
+    // simulate click of somewhere in the name of channel mention
+    const textarea = $(".o-mail-Composer-input")[0];
+    textarea.setSelectionRange(3, 3);
+    await contains(".o-mail-Composer-suggestion", { count: 1, text: "#Mario Party" });
+    // simulate click of end of the input
+    textarea.setSelectionRange(textarea.value.length, textarea.value.length);
+    await contains(".o-mail-Composer-suggestion", { count: 1, text: "Luigi" });
```

Comment:
There's no guarantee that after `await contains()`, the same actual `textarea` is used.
What I mean is, probably OWL decides to destroy this DOM element and recreate one, so `textarea` might refer to an outdated `textarea`.

It's best to use `$(".o-mail-Composer-input")[0]` again, to make sure it refers to the actual `textarea` in DOM right now.